File: fetch-directus-schema.js
#!/usr/bin/env node
/**
 * Directus Schema Analyzer
 * Fetches all collections and fields from Directus instance
 */

const https = require('https');
const fs = require('fs');

const BASE_URL = 'https://smysl-bakery-directus.onrender.com';
const TOKEN = 'Ysoj__Fjk6I8vddYHp_ZH7EbGxcf4pG8';

const options = {
  hostname: 'smysl-bakery-directus.onrender.com',
  port: 443,
  method: 'GET',
  headers: {
    'Authorization': `Bearer ${TOKEN}`,
    'Content-Type': 'application/json'
  },
  rejectUnauthorized: false // Allow self-signed certificates
};

function fetchData(path) {
  return new Promise((resolve, reject) => {
    const requestOptions = { ...options, path };
    
    const req = https.request(requestOptions, (res) => {
      let data = '';
      
      res.on('data', (chunk) => {
        data += chunk;
      });
      
      res.on('end', () => {
        try {
          const json = JSON.parse(data);
          resolve(json);
        } catch (e) {
          reject(e);
        }
      });
    });
    
    req.on('error', reject);
    req.end();
  });
}

async function main() {
  console.log('\x1b[36m🔗 Connecting to Directus...\x1b[0m');
  console.log(`URL: ${BASE_URL}\n`);
  
  try {
    console.log('\x1b[33m📚 Fetching collections...\x1b[0m');
    const collectionsData = await fetchData('/admin/api/collections');
    console.log(`\x1b[32m✅ Found ${collectionsData.data.length} collections\x1b[0m`);
    
    console.log('\x1b[33m📋 Fetching fields...\x1b[0m');
    const fieldsData = await fetchData('/api/schema/fields');
    console.log(`\x1b[32m✅ Found ${fieldsData.data.length} fields\x1b[0m\n`);
    
    // Group fields by collection
    const fieldsByCollection = {};
    fieldsData.data.forEach(field => {
      const collection = field.collection;
      if (!fieldsByCollection[collection]) {
        fieldsByCollection[collection] = [];
      }
      fieldsByCollection[collection].push(field);
    });
    
    // Sort and display report
    console.log('\x1b[36m' + '='.repeat(80) + '\x1b[0m');
    console.log('СТРУКТУРА БД DIRECTUS');
    console.log('\x1b[36m' + '='.repeat(80) + '\x1b[0m\n');
    
    let report = 'СТРУКТУРА БД DIRECTUS\n';
    report += '='.repeat(80) + '\n\n';
    
    const collections = Object.keys(fieldsByCollection).sort();
    
    for (const collectionName of collections) {
      const fields = fieldsByCollection[collectionName].sort((a, b) => 
        a.field.localeCompare(b.field)
      );
      
      console.log(`\x1b[33mКОЛЛЕКЦИЯ: ${collectionName}\x1b[0m`);
      console.log(`  Полей: ${fields.length}`);
      
      report += `КОЛЛЕКЦИЯ: ${collectionName}\n`;
      report += `  Полей: ${fields.length}\n`;
      
      fields.forEach(field => {
        console.log(`  - ${field.field} [${field.type}]`);
        report += `  - ${field.field} [${field.type}]\n`;
      });
      
      console.log();
      report += '\n';
    }
    
    // Save to file
    fs.writeFileSync('directus-schema-report.txt', report, 'utf8');
    console.log('\x1b[32m💾 Report saved to: directus-schema-report.txt\x1b[0m');
    
  } catch (error) {
    console.error(`\x1b[31m❌ Error: ${error.message}\x1b[0m`);
    process.exit(1);
  }
}

main();

File: fetch_directus_schema.py
#!/usr/bin/env python3
"""Directus Schema Analyzer"""

import requests
import json
import urllib3

# Suppress SSL warnings
urllib3.disable_warnings(urllib3.exceptions.InsecureRequestWarning)

# Configuration
BASE_URL = "https://admin.smislest.ru"
TOKEN = "Ysoj__Fjk6I8vddYHp_ZH7EbGxcf4pG8"

headers = {
    "Authorization": f"Bearer {TOKEN}",
    "Content-Type": "application/json"
}

print("🔗 Connecting to Directus...")
print(f"URL: {BASE_URL}\n")

try:
    # Get collections
    print("📚 Fetching collections...")
    collections_response = requests.get(
        f"{BASE_URL}/api/schema/collections",
        headers=headers,
        verify=False
    )
    collections_response.raise_for_status()
    collections_data = collections_response.json()
    
    # Get fields
    print("📋 Fetching fields...")
    fields_response = requests.get(
        f"{BASE_URL}/api/schema/fields",
        headers=headers,
        verify=False
    )
    fields_response.raise_for_status()
    fields_data = fields_response.json()
    
    # Group fields by collection
    fields_by_collection = {}
    for field in fields_data.get("data", []):
        collection = field.get("collection")
        if collection not in fields_by_collection:
            fields_by_collection[collection] = []
        fields_by_collection[collection].append(field)
    
    # Print report
    print("\n" + "=" * 80)
    print("СТРУКТУРА БД DIRECTUS")
    print("=" * 80 + "\n")
    
    # Sort collections
    for collection_name in sorted(fields_by_collection.keys()):
        fields = sorted(fields_by_collection[collection_name], key=lambda x: x.get("field", ""))
        
        print(f"КОЛЛЕКЦИЯ: {collection_name}")
        print(f"  Полей: {len(fields)}")
        
        for field in fields:
            field_name = field.get("field", "?")
            field_type = field.get("type", "?")
            print(f"  - {field_name} [{field_type}]")
        
        print()
    
    # Save to file
    with open("directus-schema-report.txt", "w", encoding="utf-8") as f:
        f.write("СТРУКТУРА БД DIRECTUS\n")
        f.write("=" * 80 + "\n\n")
        
        for collection_name in sorted(fields_by_collection.keys()):
            fields = sorted(fields_by_collection[collection_name], key=lambda x: x.get("field", ""))
            
            f.write(f"КОЛЛЕКЦИЯ: {collection_name}\n")
            f.write(f"  Полей: {len(fields)}\n")
            
            for field in fields:
                field_name = field.get("field", "?")
                field_type = field.get("type", "?")
                f.write(f"  - {field_name} [{field_type}]\n")
            
            f.write("\n")
    
    print("💾 Report saved to: directus-schema-report.txt")
    
except requests.exceptions.RequestException as e:
    print(f"❌ Error: {e}")
    exit(1)
